cmd/bnsd/app/msg.go

```diff
@@ -1,5 +1,19 @@
 package app
 
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/x/cash"
+)
+
+// WithFee returns a new Tx with FeeInfo set.
+func (tx Tx) WithFee(payer weave.Address, fee coin.Coin) *Tx {
```

Comment:
This is a continuation of our yesterdays' discussion on it. I know that @ethanfrey is the same opinion as you on this. Now I only have to convince both of you to change your mind :smiling_imp: 

This is returning a modified shallow copy of the original `Tx` instance. All attribute values are copied but because they are all pointers or slices in this structure changing any of them will modify both the original `Tx` and the shallow copy `Tx`. It can lead to hard to notice bugs when the same reference is modified from different objects or underlying slice memory is overwritten by appending by different `Tx` copies. :thinking:

Reply:
You are right that this is not 100% safe. I used it in scenario tests only so I did not invest into a deep clone method to do it proper. If you think about it as a builder function than this should be done proper. Will ether add clone or change the function to not return a copy.